tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -248,9 +253,21 @@ private void InstrumentationGateway_AddHeadersResponseTags(object sender, Instru
         private void Report(ITransport transport, Span span, string resultData, bool blocked)
         {
             span.SetTag(Tags.AppSecEvent, "true");
-            var samplingPirority = _settings.KeepTraces
-                                       ? SamplingPriority.UserKeep : SamplingPriority.AutoReject;
-            span.SetTraceSamplingPriority(samplingPirority);
+            var exceededTraces = _rateLimiter.UpdateTracesCounter();
+            if (exceededTraces <= 0)
+            {
+                // NOTE: DD_APPSEC_KEEP_TRACES=false means "drop all traces by setting AutoReject".
+                // It does _not_ mean "stop setting UserKeep (do nothing)". It should only be used for testing.
+                span.SetTraceSamplingPriority(_settings.KeepTraces ? SamplingPriority.UserKeep : SamplingPriority.AutoReject);
```

Comment:
The RFC says:
> the library must not modify the trace’s priority and origin fields if the priority is already strictly greater than 0

This code is not following the RFC. It's not checking what the current value if before replacing it.

Personally, I think this code is correct and the RFC is wrong, because the default value in most cases is `1` (`AutoKeep`), so if you followed the RFC you would not change that to `2` (`UserKeep`), which I think is what you want.

Reply:
I forward you comments on the RFC to the team and after a fairly length discussion we agreed what we're currently doing in the code is correct and the part of the RFC referred to will be removed.